rad/src/lib.rs

```diff
@@ -369,20 +386,127 @@ pub async fn run_retrieval(retrieve: &RADRetrieve, active_wips: ActiveWips) -> R
         retrieve,
         RadonScriptExecutionSettings::disable_all(),
         active_wips,
+        None,
     )
     .await
     .map(RadonReport::into_inner)
 }
 
+/// Run retrieval using multiple transports, and only produce a positive result if the retrieved
+/// values pass the filter function from the tally stage.
+///
+/// The idea behind this is to avoid producing commitments for data requests with sources that act
+/// in an inconsistent way, i.e. they return very different values when queried through different
+/// HTTP transports at once.
+pub async fn run_paranoid_retrieval(
+    retrieve: &RADRetrieve,
+    aggregate: RADAggregate,
+    settings: RadonScriptExecutionSettings,
+    active_wips: ActiveWips,
+    witnessing: WitnessingConfig,
+) -> Result<RadonReport<RadonTypes>> {
+    let futures: Result<Vec<_>> = witnessing
```

Comment:
This fixes the RNG issue

```suggestion
) -> Result<RadonReport<RadonTypes>> {
    if let RADType::Rng = &retrieve.kind {
        // RNG retrievals always return random values by design, so skip paranoid check
        return run_retrieval_report(retrieve, settings, active_wips, None).await;
    }
    let futures: Result<Vec<_>> = witnessing
```

Reply:
Thanks! I eventuallywent for this:

```suggestion
) -> Result<RadonReport<RadonTypes>> {
    // We can skip paranoid checks for retrieval types that don't use networking (e.g. RNG)
    if !retrieve.kind.is_http() {
        return run_retrieval_report(retrieve, settings, active_wips, None).await;
    }

    let futures: Result<Vec<_>> = witnessing
```

Reply:
Doing it that way will save us changing it in the future every time we add a new retrieval type-

Reply:
That's better, it's future-proof for new retrieval kinds.